examples/stm32/nucleo-f429zi-baremetal/Makefile

```diff
@@ -1,44 +1,60 @@
-CFLAGS  ?=  -W -Wall -Wextra -Werror -Wundef -Wshadow -Wdouble-promotion \
-						-Wformat-truncation -fno-common -Wconversion \
-						-g3 -Os -ffunction-sections -fdata-sections -I. \
-						-mcpu=cortex-m4 -mthumb -mfloat-abi=hard -mfpu=fpv4-sp-d16 $(EXTRA_CFLAGS)
-LDFLAGS ?= -Tlink.ld -nostartfiles -nostdlib --specs nano.specs -lc -lgcc -Wl,--gc-sections -Wl,-Map=$@.map
-SOURCES = boot.c main.c syscalls.c
-
-# Add Mongoose-specific flags and source files
+# Download CMSIS header files from Github on demand
+CMSIS_CORE_VERSION ?= 5.9.0      # ARM Cortex-M definitions
+CMSIS_CORE_REPO ?= https://github.com/ARM-software/CMSIS_5
+CMSIS_DEVICE_VERSION ?= v2.6.8   # ST MCU peripheral definitions
+CMSIS_DEVICE_REPO ?= https://github.com/STMicroelectronics/cmsis_device_f4
+
+CFLAGS ?= -W -Wall -Wextra -Werror -Wundef -Wshadow -Wdouble-promotion \
+          -Wformat-truncation -fno-common -Wconversion \
+          -g3 -Os -ffunction-sections -fdata-sections \
+          -I . -I cmsis_core/CMSIS/Core/Include -I cmsis_device_f4/Include \
+          -mcpu=cortex-m4 -mthumb -mfloat-abi=hard -mfpu=fpv4-sp-d16 $(EXTRA_CFLAGS)
+LDFLAGS ?= -Tlink.ld -nostdlib -nostartfiles --specs nano.specs -lc -lgcc -Wl,--gc-sections -Wl,-Map=$@.map
+SOURCES = main.c syscalls.c cmsis_device_f4/Source/Templates/gcc/startup_stm32f429xx.s
```

Comment:
We are introducing assembly files, syntax changes between compilers, we should perhaps add this in a different line with a comment; I mean, if this is going to be a skeleton

Reply:
Good point too! Will fix in my next PR.